SpatialGDK/Source/SpatialGDKTests/Public/GDKAutomationTestBase.h

```diff
@@ -0,0 +1,108 @@
+﻿// Copyright (c) Improbable Worlds Ltd, All Rights Reserved
+
+#pragma once
+#include "Misc/AutomationTest.h"
+#include "SpatialGDKTests/SpatialGDKServices/LocalDeploymentManager/LocalDeploymentManagerUtilities.h"
+
+DEFINE_LOG_CATEGORY_STATIC(LogGDKTestBase, Log, All);
+
+/**
+ * This class extends the Unreal AutomationTestBase to allow unit tests to be augmented with a set-up step, called
+ * before each test.
+ * This class is then offered through a macro, in a similar way to `IMPLEMENT_SIMPLE_AUTOMATION_TEST`.
+ *
+ * To use this test base, the GDK_AUTOMATION_TEST macro should be used, followed by the test body:
+ * ```
+ *	   GDK_AUTOMATION_TEST(MyModule, MyComponent, MyTestName)
+ *	   {
+ *			// do some testing here...
+ *
+ *			return true;
+ *	   }
+ * ```
+ *
+ * Returning `true` indicates a test pass and returning `false` indicates test failure.
+ */
+class FGDKAutomationTestBase : public FAutomationTestBase
+{
+public:
+	FGDKAutomationTestBase(const FString& Name, bool bInComplexTask, FString TestSrcFileName, uint32 TestSrcFileLine)
+		: FAutomationTestBase(Name, bInComplexTask)
+		, TestName(Name)
+		, TestSourceFileName(TestSrcFileName)
+		, TestSourceFileLine(TestSrcFileLine)
+	{
+	}
+
+	virtual uint32 GetTestFlags() const override
+	{
+		return EAutomationTestFlags::ApplicationContextMask | EAutomationTestFlags::EngineFilter;
+	}
+
+	virtual bool IsStressTest() { return false; }
+	virtual uint32 GetRequiredDeviceNum() const override { return 1; }
+
+protected:
+	/**
+	 * Checks for an existing deployment and stops it if one exists (also killing the associated workers).
+	 * Ran before each test.
+	 */
+	virtual void SetUp()
+	{
+		FLocalDeploymentManager* LocalDeploymentManager = SpatialGDK::GetLocalDeploymentManager();
+		if (LocalDeploymentManager->IsLocalDeploymentRunning() || LocalDeploymentManager->IsDeploymentStarting())
+		{
+			UE_LOG(LogGDKTestBase, Warning, TEXT("Deployment found! (Was this left over from another test?)"))
```

Comment:
Would this not get picked up by the automation system as failures? In 4.26 we observed that with functional tests, any warnings printed during the test execution would be treated as the test failure, not sure if it would be the same for regular unit tests.

Reply:
Unit tests can have warnings without failing, they just get flagged up in [the test output](https://buildkite-artifacts.production-intinf-eu1.i8e.io/organizations/improbable/pipelines/unrealgdk-premerge/builds/3f7a745a-742b-4772-935c-1210168facba/jobs/v4-20-11-18-224740-bk17641-0c4125be-d/854692a7-ee69-4fa7-8c68-710063793895/windows/b%5C1%5Cunrealgdk-premerge%5Cci%5CGDKTestGyms%5CTestResults%5Cindex.html)

I think this is something we want to be aware of like this with a bit more emphasis than just a normal log which gets buried

I guess if 4.26 changes the semantics of a Warning to mean unit test failure, this can be fixed pretty easily

Reply:
that will be interesting. if that is the case many tests will fail on mac with 4.26 as macos prints some additional warnings

Reply:
but no, atm (with 4.25) warnings are not failures afaik

Reply:
I can confirm that these warnings don't fail the tests in 4.26.